addons/mail/models/mail_link_preview.py

```diff
@@ -68,15 +62,22 @@ def _is_link_preview_enabled(self):
         return link_preview_throttle > 0
 
     @api.model
-    def _is_domain_throttled(self, url):
-        domain = urlparse(url).netloc
-        date_interval = fields.Datetime.to_string((datetime.now() - relativedelta(seconds=10)))
-        call_counter = self.search_count([
-            ('source_url', 'ilike', domain),
-            ('create_date', '>', date_interval),
-        ])
-        link_preview_throttle = int(self.env['ir.config_parameter'].get_param('mail.link_preview_throttle', 99))
-        return call_counter > link_preview_throttle
+    def _get_link_preview_from_url(self, url):
+        """Return the URL preview, first from the database if available otherwise make the request."""
+        lifetime = int(self.env['ir.config_parameter'].sudo().get_param('mail.link_preview_lifetime', 3))
```

Comment:
Maybe (use full model name + precise lifetime timespan)
```suggestion
        lifetime = int(self.env['ir.config_parameter'].sudo().get_param('mail.mail_link_preview_lifetime_days', 3))
```